src/main/java/io/neonbee/entity/EntityModelManager.java

```diff
@@ -357,102 +230,155 @@ public static Future<Map<String, EntityModel>> reloadModels(NeonBee neonBee) {
-        return registerModels(NeonBee.get(vertx), EXTERNAL_MODEL_IDENTIFIERS
-                .computeIfAbsent(NeonBee.get(vertx), Functions.forSupplier(ConcurrentHashMap::new))
-                .computeIfAbsent(modelIdentifier,
-                        newIdentifier -> new EntityModelDefinition(csnModelDefinitions, associatedModelDefinitions)));
+    public Future<Map<String, EntityModel>> registerModels(EntityModelDefinition modelDefinition) {
+        return externalModelDefinitions.add(modelDefinition) ? reloadModels() : succeededFuture(getBufferedModels());
     }
 
     /**
-     * Register an external model. External models will be loaded alongside any models that are in the class path or in
-     * the models directory.
+     * Unregisters a given external model and reload the models.
      *
-     * @see #registerModels(EntityModelDefinition)
-     * @deprecated use {@code neonBee.getModelManager().registerModels(modelDefinition)} instead
-     * @param neonBee         the {@link NeonBee} instance
-     * @param modelDefinition the entity model definition
+     * @param modelDefinition the model definition to remove
      * @return a {@link Future} to a map from schema namespace to EntityModel
      */
-    @Deprecated(forRemoval = true)
-    public static Future<Map<String, EntityModel>> registerModels(NeonBee neonBee,
-            EntityModelDefinition modelDefinition) {
-        return neonBee.getModelManager().registerModels(modelDefinition);
+    public Future<Map<String, EntityModel>> unregisterModels(EntityModelDefinition modelDefinition) {
+        return externalModelDefinitions.remove(modelDefinition) ? reloadModels() : succeededFuture(getBufferedModels());
     }
 
     /**
-     * Register an external model. External models will be loaded alongside any models that are in the class path or in
-     * the models directory.
+     * copies changed model files to distributed map in cluster.
      *
-     * @param modelDefinition the entity model definition
-     * @return a {@link Future} to a map from schema namespace to EntityModel
+     * @return future
      */
-    public Future<Map<String, EntityModel>> registerModels(EntityModelDefinition modelDefinition) {
-        return externalModelDefinitions.add(modelDefinition) ? reloadModels() : getSharedModels();
+    public Future<Void> pushChanges(Vertx vertx) {
+        LOGGER.info("pushing entity models changes in shared memory");
+        final List<Future> futures = new ArrayList<>(externalModelDefinitions.size());
+        return getLock(vertx).compose(lock -> getAsyncMap(vertx).compose(m -> {
+            externalModelDefinitions.forEach(e -> futures.add(m.putIfAbsent(e.toBuffer(), getNodeId(vertx))));
+            LOGGER.info("entity models written into shared memory by node id " + getNodeId(vertx) + " = "
+                    + externalModelDefinitions);
+            return CompositeFuture.all(futures).compose(result -> {
```

Comment:
Shouldn't we use join instead of all to wait for all futures even if a future fails?